kernel/sched.c

```diff
@@ -1279,6 +1279,9 @@ static inline void z_vrfy_k_thread_priority_set(k_tid_t thread, int prio)
 #ifdef CONFIG_SCHED_DEADLINE
 void z_impl_k_thread_deadline_set(k_tid_t tid, int deadline)
 {
+
+	__ASSERT(deadline > 0, "The deadline should be larger than zero.");
+
```

Comment:
@TaiJuWu - the `CLAMP()` macro restricts a value to a range. 
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util.h#L395

Reply:
> @TaiJuWu - the `CLAMP()` macro restricts a `val` to the range `[low, high]` (inclusive). https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util.h#L395

Thanks for your answer.